src/intelligence_layer/connectors/argilla/argilla_wrapper_client.py

```diff
@@ -197,11 +204,14 @@ def ensure_workspace_exists(self, workspace_name: str) -> str:
         Returns:
             The name of an argilla workspace with the given `workspace_name`.
         """
-        try:
-            workspace = rg.Workspace.from_name(workspace_name)
-            return str(workspace.name)
-        except ValueError:
-            return str(rg.Workspace.create(name=workspace_name).name)
+        workspace = self.client.workspaces(name=workspace_name)
+        if workspace:
+            return workspace_name
+
+        workspace = rg.Workspace(name=workspace_name, client=self.client)
+        workspace.create()
+        assert workspace.name
```

Comment:
maybe with raise and a somewhat descriptive error message instead of assert? I guess this should never happen regardless

Reply:
Yes, very good idea!